### ISSUE TITLE
[Bug] Talkback doesn't explicitly announce aria-disabled `li` element 

### ISSUE THREAD

**Issue by LFDanLu** [mozilla-mobile/fenix#17700]
## Steps to reproduce
Precondition: Using Firefox for Android
Combo box example:
1. Navigate to https://lby61.csb.app/
2. With Talkback enabled, focus the input field's arrow button and double tap
3. When the menu opens, focus the "Cat" option. Note that it only announces "Cat" despite it being aria-disabled

Simple example:
1. Navigate to https://jzy9y.csb.app/
2. With Talkback enabled, focus "Test". Note that it doesn't announce that the `li` is disabled despite it being aria-disabled
3. Focus the input field beneath "Test". Note that it properly announces "Entry disabled".

### Expected behavior
Would be nice if Talkback on Firefox for Android explicitly announced `li` (and other elements) as disabled if aria-disabled is applied to them. Would establish behavioral parity with Chrome and Samsung Internet on Android.

NB: enabled `li` options in the combo box will announce with "double tap to activate" so a Talkback user can technically distinguish between the an enabled and disabled option. Its a bit problematic still IMO since non-clickable section headers in the combobox can't be distinguished from disabled items since Talkback only announces the text for both.   

### Actual behavior
Talkback on Firefox for Android doesn't explicitly announce that a `li` element are disabled.

### Device information

* Device vendor / model and Android version: Android 10, Galaxy S9+
* Firefox for Android version: 85.1.1

**Comment by ekager** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1689718

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
